xbmc/settings/AdvancedSettings.h

```diff
@@ -386,4 +394,7 @@ class CAdvancedSettings : public ISettingCallback, public ISettingsHandler
     void Initialize();
     void Clear();
     void SetExtraArtwork(const TiXmlElement* arttypes, std::vector<std::string>& artworkMap);
+
+    mutable CCriticalSection m_critSection; //!< protects the callbacks list access
```

Comment:
If this mutex is only for protecting the callbacks list, like the comment suggests, you probably should name the member accordingly, then the comment would even not be needed anymore.

Reply:
Yes, done. Access to a lot of class members should probably be protected but that would be best done as a dedicated PR. Rarely if ever is a problem because settings are typically loaded once on start and there is no uncontrolled activity. Profile switch may be more risky.